### PR TITLE
Get list of file urls for uploaded files from azure provider

### PR DESCRIPTION
**Pull request by bishal-pdMSFT**

### CODE REVIEW

File: Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts [hunk 1]
```diff
@@ -30,8 +29,8 @@ export class AzureBlobProvider implements models.IArtifactProvider {
                 } else {
                     var blobUrl = self._blobSvc.getUrl(self._container, blobPath);
                     console.log(tl.loc("CreatedBlobForItem", item.path, blobUrl));
-                    newArtifactItem.metadata["downloadUrl"] = blobUrl;
-                    resolve(newArtifactItem);
+                    item.metadata["destinationUrl"] = blobUrl;
+                    resolve(item);
```
Comment (sachinma) [inline Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts]: You are resolving here; but wont the upload complete much later. Or is that handled correctly

Comment (sachinma) [reply Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts]: I am not sure of the intended return type: Should it Should we wait for the pipe to finish before returning the copied URL.

File: Tasks/Common/azure-blobstorage-artifactProvider/blobservice.ts [hunk 2]
```diff
@@ -13,11 +14,21 @@ export class BlobService {
         this._storageAccessKey = storageAccessKey;
     }
 
-    public async uploadBlobs(source: string, container: string, prefixFolderPath?: string, host?: string): Promise<void> {
+    public async uploadBlobs(source: string, container: string, prefixFolderPath?: string, host?: string): Promise<string[]> {
```
Comment (sachinma) [inline Tasks/Common/azure-blobstorage-artifactProvider/blobservice.ts]: what is host? supposed to be

File: Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts (hunk 1 above)
Comment (bishal-pdMSFT) [reply Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts]: callback is called only after writeStream's 'end' event.

File: Tasks/Common/azure-blobstorage-artifactProvider/blobservice.ts (hunk 2 above)
Comment (bishal-pdMSFT) [reply Tasks/Common/azure-blobstorage-artifactProvider/blobservice.ts]: I will change this to be taken from VMSS task

**Pull request merged**